plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleConnectorTest.java

```diff
@@ -496,6 +501,172 @@ private void predicatePushdownTest(String oracleType, String oracleLiteral, Stri
+                }
+            }
+
+            // nested cast
+            CastTestCase testCase = new CastTestCase("varchar(10)", "varchar");
+            try (TestTable tableA = new TestTable(getQueryRunner()::execute, "test_join_cast_a_", "(a1 %s)".formatted(testCase.fromType()), asList("('India')", "('Poland')", "(null)"));
+                    TestTable tableB = new TestTable(getQueryRunner()::execute, "test_join_cast_b_", "(b1 %s)".formatted(testCase.targetType()), asList("('India')", "('France')", "(null)"))) {
+                assertJoinFullyPushedDown(session, tableA.getName(), tableB.getName(), "CAST(CAST(l.a1 AS %s) AS VARCHAR(50)) = r.b1".formatted(testCase.castType()));
+            }
+
+            // Different joins
+            try (TestTable leftTable = new TestTable(getQueryRunner()::execute, "left_table_", "(id int, varchar_50 varchar(50))", asList("(1, 'India')", "(2, 'Poland')", "(3, null)"));
+                    TestTable rightTable = new TestTable(getQueryRunner()::execute, "right_table_", "(varchar_100 varchar(100))", asList("('India')", "('France')", "(null)"))) {
+                String leftTableName = leftTable.getName();
+                String rightTableName = rightTable.getName();
+
+                assertJoinFullyPushedDown(session, leftTableName, rightTableName, "LEFT JOIN", "CAST(l.varchar_50 AS VARCHAR(100)) = r.varchar_100", asList(1, 2, 3));
+                assertJoinFullyPushedDown(session, leftTableName, rightTableName, "LEFT JOIN", "l.varchar_50 = CAST(r.varchar_100 AS VARCHAR(50))", asList(1, 2, 3));
+                assertJoinFullyPushedDown(session, leftTableName, rightTableName, "LEFT JOIN", "CAST(l.varchar_50 AS VARCHAR(200)) = CAST(r.varchar_100 AS VARCHAR(200))", asList(1, 2, 3));
+
+                assertJoinFullyPushedDown(session, leftTableName, rightTableName, "RIGHT JOIN", "CAST(l.varchar_50 AS VARCHAR(100)) = r.varchar_100", asList(1, null, null));
+                assertJoinFullyPushedDown(session, leftTableName, rightTableName, "RIGHT JOIN", "l.varchar_50 = CAST(r.varchar_100 AS VARCHAR(50))", asList(1, null, null));
+                assertJoinFullyPushedDown(session, leftTableName, rightTableName, "RIGHT JOIN", "CAST(l.varchar_50 AS VARCHAR(200)) = CAST(r.varchar_100 AS VARCHAR(200))", asList(1, null, null));
+
+                assertJoinFullyPushedDown(session, leftTableName, rightTableName, "INNER JOIN", "CAST(l.varchar_50 AS VARCHAR(100)) = r.varchar_100", List.of(1));
+                assertJoinFullyPushedDown(session, leftTableName, rightTableName, "INNER JOIN", "l.varchar_50 = CAST(r.varchar_100 AS VARCHAR(50))", List.of(1));
+                assertJoinFullyPushedDown(session, leftTableName, rightTableName, "INNER JOIN", "CAST(l.varchar_50 AS VARCHAR(200)) = CAST(r.varchar_100 AS VARCHAR(200))", List.of(1));
+            }
+        }
+
+        private static List<CastTestCase> supportedCastTypePushdown()
```

Comment:
From the rule it looked like `integer` to `varchar` or `char` looks like a supported cast pushdown - Can we restrict them or add test cases here ?